herddb-core/src/main/java/herddb/client/HDBConnection.java

```diff
@@ -415,7 +415,7 @@ private RoutedClientSideConnection getRouteToServer(String nodeId) throws Client
         }
     }
 
-    private RoutedClientSideConnection getRouteToTableSpace(String tableSpace) throws ClientSideMetadataProviderException, HDBException {
+    public RoutedClientSideConnection getRouteToTableSpace(String tableSpace) throws ClientSideMetadataProviderException, HDBException {
```

Comment:
this should not  be exposed to the public API, or at least we should add a javadoc that tells not to use it

Reply:
Reduced to protected and handled public visibility in test